private/rules/pom_file.bzl

```diff
@@ -1,33 +1,41 @@
 load(":has_maven_deps.bzl", "MavenInfo", "has_maven_deps")
 
-_PLAIN_DEP = """        <dependency>
-            <groupId>{0}</groupId>
-            <artifactId>{1}</artifactId>
-            <version>{2}</version>
-        </dependency>"""
+def _format_dep(unpacked):
```

Comment:
I know why this always creates the longer version of the dependency declaration (because it's consistent and that's always nice), but it reads terribly in the generated POM files, which is why we had the two alternatives before.

Perhaps this method could conditionally add the `type` and `scope` fields if they need to be added?

We can address this in a follow up PR, but it'd be nice to do it now if you can.

Reply:
that's actually what it tries to do, maybe in a little sneaky way by having
```
       ("            <type>%s</type>\n" % unpacked.type) if unpacked.type and unpacked.type != "jar" else "",
       ("            <scope>%s</scope>\n" % unpacked.scope) if unpacked.scope and unpacked.scope != "compile" else "",
```
so lines should be absent when they aren't needed, but there could be a cleaner way to achieve that